gtk+3: Fix deps and test for Linuxbrew

maxim-belkin opened the pull request:
* Depends on atk-spi-2, fontconfig, and harfbuzz
* The test used the Mac-only -lintl

sjackman commented:
TravisCI
```
pango-engine.h:335: Warning: Pango: symbol='script_engine_create': Unknown namespace for symbol 'script_engine_create'
Package xcb-shm was not found in the pkg-config search path.
Perhaps you should add the directory containing `xcb-shm.pc'
to the PKG_CONFIG_PATH environment variable
Package 'xcb-shm', required by 'cairo', not found
/tmp/pango-20170415-5979-1dqfo0/pango-1.40.5/pango/tmp-introspectU0zMz8/Pango-1.0.c:2:18: fatal error: glib.h: No such file or directory
compilation terminated.
Traceback (most recent call last):
  File "/home/linuxbrew/.linuxbrew/Cellar/gobject-introspection/1.52.0_1/bin/g-ir-scanner", line 66, in <module>
    sys.exit(scanner_main(sys.argv))
  File "/home/linuxbrew/.linuxbrew/Cellar/gobject-introspection/1.52.0_1/lib/gobject-introspection/giscanner/scannermain.py", line 547, in scanner_main
```
https://s3.amazonaws.com/archive.travis-ci.org/jobs/222378437/log.txt

maxim-belkin commented:
This will be fixed by #2302 (if merged)

sjackman commented:
Let's merge #2302.

maxim-belkin commented:
rebased on top of #2302

rwhogg commented:
Looks like we have to fix `pango` first too.

sjackman commented:
Blocked by `pango` #2334

maxim-belkin commented:
```
Package 'xrender', required by 'cairo', not found
```

maxim-belkin commented:
so, we need `linuxbrew/xorg/libxrender`

maxim-belkin commented:
initiated a rebuild on Circle because the failed part was:
```
Download failed: https://linuxbrew.bintray.com/bottles/gcc-5.3.0.x86_64_linux.bottle.tar.gz
==> FAILED
```

maxim-belkin commented:
Success!!!!!

maxim-belkin commented:
But TestBot hasn't created a release 😠

maxim-belkin commented:
rebased to see if it helps

sjackman commented:
Ping me if the rebase doesn't help.

maxim-belkin commented:
ping (still no release here: https://github.com/LinuxbrewTestBot/homebrew-core/releases/tag/pr-2368)

sjackman commented:
Uh oh. I think it may be a CircleCI bug, possibly due to the `+` in the file name? I don't think that was a problem before.
Go to https://circleci.com/gh/Linuxbrew/homebrew-core/5470#artifacts/containers/0
Click on [`gtk%2B3-3.22.11.x86_64_linux.bottle.json`](https://5470-56942570-gh.circle-artifacts.com/0/tmp/circle-artifacts.0FnhecN/gtk%252B3-3.22.11.x86_64_linux.bottle.json)
produces `Error loading artifact!`
@maxim-belkin Can you report an issue with CircleCI?

maxim-belkin commented:
done https://discuss.circleci.com/t/artifacts-with-sign-in-their-name/12280

sjackman commented:
It may or may not be the plus sign that's causing the issue. That was just a guess. `gtk+` has a bottle, so that may not be the issue.

maxim-belkin commented:
they [responded](https://discuss.circleci.com/t/artifacts-with-special-characters-in-the-file-name/12280/2):

> We are aware of this issue and should have shipped a fix for this. Can you let me know if this is still a problem for newer builds?

maxim-belkin commented:
@sjackman 
the file created by Circle had a bad name: "+" became "%252B3"

Pulled bottle manually with
```
brew pull-circle --ci-upload https://github.com/Linuxbrew/homebrew-core/pull/2368
```

renamed the file:
```
mv gtk%252B3-3.22.11.x86_64_linux.bottle.tar.gz gtk+3-3.22.11.x86_64_linux.bottle.tar.gz
```

extracted sha256 from `gtk%252B3-3.22.11.x86_64_linux.bottle.json`
Manually edited `gtk+3` formula to add sha256
Ran `brew audit --strict gtk+3`

Ran `brew install gtk+3` to test 🤕 
```
$ brew install gtk+3
==> Installing dependencies for gtk+3: linuxbrew/xorg/xtrans, linuxbrew/xorg/bigreqsproto, linuxbrew/xorg/compositeproto, linuxbrew/xorg/damageproto, linuxbrew/xorg/dmxproto, linuxbrew/xorg/dri2proto, linuxbrew/xorg/dri3proto, linuxbrew/xorg/fixesproto, linuxbrew/xorg/fontsproto, linuxbrew/xorg/glproto, linuxbrew/xorg/inputproto, linuxbrew/xorg/kbproto, linuxbrew/xorg/presentproto, linuxbrew/xorg/randrproto, linuxbrew/xorg/recordproto, linuxbrew/xorg/renderproto, linuxbrew/xorg/resourceproto, linuxbrew/xorg/scrnsaverproto, linuxbrew/xorg/videoproto, linuxbrew/xorg/xcmiscproto, linuxbrew/xorg/xextproto, linuxbrew/xorg/xf86bigfontproto, linuxbrew/xorg/xf86dgaproto, linuxbrew/xorg/xf86driproto, linuxbrew/xorg/xf86vidmodeproto, linuxbrew/xorg/xineramaproto, linuxbrew/xorg/xproto, linuxbrew/xorg/xorg-protocols, linuxbrew/xorg/libxau, linuxbrew/xorg/libpthread-stubs, linuxbrew/xorg/libxdmcp, linuxbrew/xorg/libxcb, linuxbrew/xorg/libx11, linuxbrew/xorg/libxext, linuxbrew/xorg/libfs, linuxbrew/xorg/libice, linuxbrew/xorg/libsm, linuxbrew/xorg/libxscrnsaver, linuxbrew/xorg/libxt, linuxbrew/xorg/libxmu, linuxbrew/xorg/libxpm, linuxbrew/xorg/libxaw, linuxbrew/xorg/libxfixes, linuxbrew/xorg/libxcomposite, linuxbrew/xorg/libxrender, linuxbrew/xorg/libxcursor, linuxbrew/xorg/libxdamage, linuxbrew/xorg/libfontenc, libpng, bzip2, freetype, linuxbrew/xorg/libxfont, expat, fontconfig, linuxbrew/xorg/libxft, linuxbrew/xorg/libxi, linuxbrew/xorg/libxinerama, linuxbrew/xorg/libxrandr, linuxbrew/xorg/libxres, linuxbrew/xorg/libxtst, linuxbrew/xorg/libxv, linuxbrew/xorg/libxvmc, linuxbrew/xorg/libxxf86dga, linuxbrew/xorg/libxxf86vm, linuxbrew/xorg/libdmx, linuxbrew/xorg/libpciaccess, linuxbrew/xorg/libxkbfile, linuxbrew/xorg/libxshmfence, linuxbrew/xorg/xorg, ncurses, readline, sqlite, gdbm, berkeley-db@4, python, libxml2, gettext, libffi, pcre, util-linux, glib, jpeg, libtiff, pkg-config, pixman, cairo, m4, bison, flex, gobject-introspection, shared-mime-info, gdk-pixbuf, atk, linuxbrew/xorg/libdrm, libcap, systemd, linuxbrew/xorg/libsha1, gmp, mpfr, libmpc, isl, gcc, binutils, libedit, llvm, libelf, linuxbrew/xorg/libomxil-bellagio, linuxbrew/xorg/wayland, valgrind, linuxbrew/xorg/libva, linuxbrew/xorg/libvdpau, linuxbrew/xorg/mesa, libepoxy, icu4c, harfbuzz, pango, hicolor-icon-theme, gsettings-desktop-schemas
```

committed to local master branch
and pushing it now

maxim-belkin commented:
Darn:
```
$ brew linkage gtk+3
System libraries:
  /lib/x86_64-linux-gnu/libc.so.6
  /lib/x86_64-linux-gnu/libm.so.6
  /lib/x86_64-linux-gnu/libpthread.so.0
  /lib/x86_64-linux-gnu/librt.so.1
Homebrew libraries:
  /home/linuxbrew/.linuxbrew/lib/libatk-1.0.so.0 (atk)
  /home/linuxbrew/.linuxbrew/lib/libcairo-gobject.so.2 (cairo)
  /home/linuxbrew/.linuxbrew/lib/libcairo.so.2 (cairo)
  /home/linuxbrew/.linuxbrew/lib/libfontconfig.so.1 (fontconfig)
  /home/linuxbrew/.linuxbrew/lib/libfreetype.so.6 (freetype)
  /home/linuxbrew/.linuxbrew/lib/libgdk_pixbuf-2.0.so.0 (gdk-pixbuf)
  /home/linuxbrew/.linuxbrew/lib/libgio-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/lib/libglib-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/lib/libgmodule-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/lib/libgobject-2.0.so.0 (glib)
  /home/linuxbrew/.linuxbrew/lib/libgdk-3.so.0 (gtk+3)
  /home/linuxbrew/.linuxbrew/lib/libgtk-3.so.0 (gtk+3)
  /home/linuxbrew/.linuxbrew/lib/libharfbuzz.so.0 (harfbuzz)
  /home/linuxbrew/.linuxbrew/lib/libepoxy.so.0 (libepoxy)
  /home/linuxbrew/.linuxbrew/lib/libX11.so.6 (linuxbrew/xorg/libx11)
  /home/linuxbrew/.linuxbrew/lib/libXcomposite.so.1 (linuxbrew/xorg/libxcomposite)
  /home/linuxbrew/.linuxbrew/lib/libXcursor.so.1 (linuxbrew/xorg/libxcursor)
  /home/linuxbrew/.linuxbrew/lib/libXdamage.so.1 (linuxbrew/xorg/libxdamage)
  /home/linuxbrew/.linuxbrew/lib/libXext.so.6 (linuxbrew/xorg/libxext)
  /home/linuxbrew/.linuxbrew/lib/libXfixes.so.3 (linuxbrew/xorg/libxfixes)
  /home/linuxbrew/.linuxbrew/lib/libXi.so.6 (linuxbrew/xorg/libxi)
  /home/linuxbrew/.linuxbrew/lib/libXinerama.so.1 (linuxbrew/xorg/libxinerama)
  /home/linuxbrew/.linuxbrew/lib/libXrandr.so.2 (linuxbrew/xorg/libxrandr)
  /home/linuxbrew/.linuxbrew/lib/libpango-1.0.so.0 (pango)
  /home/linuxbrew/.linuxbrew/lib/libpangocairo-1.0.so.0 (pango)
  /home/linuxbrew/.linuxbrew/lib/libpangoft2-1.0.so.0 (pango)
Missing libraries:
  libatk-bridge-2.0.so.0
Possible undeclared dependencies:
  fontconfig
  freetype
  harfbuzz
  linuxbrew/xorg/libx11
  linuxbrew/xorg/libxcomposite
  linuxbrew/xorg/libxcursor
  linuxbrew/xorg/libxdamage
  linuxbrew/xorg/libxext
  linuxbrew/xorg/libxfixes
  linuxbrew/xorg/libxi
  linuxbrew/xorg/libxinerama
  linuxbrew/xorg/libxrandr
```

sjackman commented:
```
Missing libraries:
  libatk-bridge-2.0.so.0
```
On Debian this file is provided by the source package `at-spi2-atk` and binary packages `libatk-bridge2.0-0` and `libatk-bridge2.0-dev`.
https://packages.debian.org/source/stretch/at-spi2-atk
